structs.go

```diff
@@ -14,6 +14,10 @@ var (
 	DefaultTagName = "structs" // struct's field default tag name
 )
 
+type Mapper interface {
+	Map(TagName string) map[string]interface{}
+}
+
```

Comment:
Currently it seems we need the TagName so the mapper implementation can call `structs` package and pass on the custom `TagName`.

A typical function implementing this mapper would be:
```
func (s *foo) Map(TagName string) map[string]interface{} {
    m := structs.New(s.data)
    m.TagName = TagName
    return m.Map()
}
```

Reply:
Maybe we could create the following function on the package to simplify this:
```
func MapWithTag(s interface{}, TagName string) map[string]interface{} {
    m := New(s)
    m.TagName = TagName
    return m.Map()
}
```
Then a function implementing the Mapper interface would be:
```
func (s *foo) Map(TagName string) map[string]interface{} {
    return structs.MapWithTag(s.data, TagName)
}
```